src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -345,10 +346,17 @@ private void sendDatasetApprovalNotificationEmail(Dataset dataset, User user, Bo
           dac.getName(),
           dataset.getDatasetIdentifier());
     } else {
-      emailService.sendDatasetDeniedMessage(
-          user,
-          dac.getName(),
-          dataset.getDatasetIdentifier());
+      if (!Objects.isNull(dac.getEmail())) {
```

Comment:
It looks this use of `Objects.isNull()` is common in this codebase, and it works, but it wasn't intended for this use. It was intended to be used as a predicate, such as for a streaming operation. See https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Objects.html#isNull(java.lang.Object). Personally I would compare against `null` directly instead:

```suggestion
      if (dac.getEmail() != null) {
```

If you do want to use an `Objects` method, I would use `nonNull()` instead as then you don't have to negate the result:

```suggestion
      if (Objects.nonNull(dac.getEmail())) {
```

Reply:
Thanks for the link to the doc, don't think I've ever read that. I've been suggesting the `Objects` methods for some time now.

Reply:
If you'd rather not compare against `null` directly, a lot of people like using `Optional.ofNullable()` instead. So you'd write something like:
```
     Optional.ofNullable(dac.getEmail()).ifPresentOrElse(dacEmail ->
        emailService.sendDatasetDeniedMessage(
            user,
            dac.getName(),
            dataset.getDatasetIdentifier(),
            dacEmail)).orElse(),
        () -> logWarn("Unable to send dataset denied email to DAC: " + dac.getDacId());
```